### PR TITLE
Proposal to implement `attrs` support

### PR DESCRIPTION
**Pull request by sasanjac**
- implemented a method that compiles imports in regard to `attrs` including their import alias
- check at class definitions if any decorator is one of `attrs` dataclass decorators (matched against the actual import alias)

### CODE REVIEW

**Comment by sondrelg** [comment]
Thanks @sasanjac. I'll try myst best to review this tomorrow 👍

**Comment by sondrelg** [comment]
Code looks solid @sasanjac. Thanks for taking the time to add this 🚀

File: tests/test_attrs.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+def test_non_attrs_model(expected):
+    example = textwrap.dedent(
+        '''
+        from decimal import Decimal
+
+        class X:
+            x: Decimal
+        '''
+    )
+    assert _get_error(example, error_code_filter='TC001,TC002') == expected
```
Comment (sondrelg) [inline tests/test_attrs.py:32]: Are we not just running the same test twice here?

```suggestion
def test_non_attrs_model(expected):
    example = textwrap.dedent(
        '''
        from decimal import Decimal

        class X:
            x: Decimal
        '''
    )
    assert _get_error(example, error_code_filter='TC001,TC002') == {'2:0 ' + TC002.format(module='decimal.Decimal')}
```

File: tests/test_attrs.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        ["import attrs", "@attrs.define"],
+        ["import attr", "@attr.s(auto_attribs=True)"],
+        ["import attr", "@attr.define"],
+    ),
+)
+def test_attrs_model(imp, dec):
+    """
+    A class cannot be a attrs model if it doesn't have a base class,
+    so we should raise the same error here in both cases.
+    """
```
Comment (sondrelg) [inline tests/test_attrs.py:46]: Don't think this or the other 5 docstrings are up to date 🙂 Can you update them to explain the purpose of the test?

**Review by sondrelg — CHANGES_REQUESTED** [review]
Please see comments 👍 Code looks good, but think tests need some small cleanups.

Also, to run linters, please run:
```
pip install pre-commit
pre-commit install
pre-commit run --all-files
```

When that is done I'll merge and release a new minor version 👏

File: tests/test_attrs.py (hunk 1 above)
Comment (sasanjac) [reply tests/test_attrs.py:32]: True, it was for my internal testing that I don't break functionality unrelated to `attrs`.

File: tests/test_attrs.py (hunk 2 above)
Comment (sasanjac) [reply tests/test_attrs.py:46]: Sorry, totally forgot about that.

**Comment by sasanjac** [comment]
I had to do some config changes:
- `flake8` fails due to the use of `typing.Any` (code ANN401) which comes from upstream code, so I had to ignore the error instead of refactoring unrelated code
- `black` fails due to breaking changes in `click` (https://github.com/psf/black/issues/2964), so I had to pin the version for now

**Comment by sondrelg** [comment]
Really nice @sasanjac 👏  I'll release this now

**Comment by sondrelg** [comment]
Looks like Github isn't attributing the commits to your profile @sasanjac 

image

Do you know why? If there's something I can do to remedy it, let me know 👍

**Comment by sasanjac** [comment]
Hmm, I don't know either. Maybe because the commits were made less than 24 hours ago, according to Github Docs or because I move forks to an organization to keep my profile tidy 🤷‍♂️
I'll check again in 24 hours

**Comment by sasanjac** [comment]
It was just an email address mismatch 😅

**Pull request merged**